src/features/TakeNow/ui/TakeNowSuccessModal.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState } from 'react';
+
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { MuiModal } from '~/shared/ui';
+import { useCustomTranslation, useOnceEffect } from '~/shared/utils';
+
+export const TakeNowSuccessModal = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { t } = useCustomTranslation();
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  useOnceEffect(() => {
+    if (location.state && location.state.showTakeNowSuccessModal && !isOpen) {
+      navigate(window.location.pathname, {
+        state: {
+          ...location.state,
+          showTakeNowSuccessModal: undefined,
+        } as unknown,
+        replace: true,
+      });
+
+      setIsOpen(true);
+    }
+  });
+
+  return (
+    <MuiModal
+      isOpen={isOpen}
+      onHide={() => {}}
```

Comment:
The underlying `onClose` prop accepts `undefined` here to avoid the lint warning about an empty arrow function

Reply:
Good catch. I forgot that I was going to make this optional